url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_SelectClauseParser.java

```diff
@@ -0,0 +1,7251 @@
+		RewriteRuleSubtreeStream stream_selectItem=new RewriteRuleSubtreeStream(adaptor,"rule selectItem");
+
+		 gParent.msgs.push("select list"); 
+		try {
+			// SelectClauseParser.g:68:5: ( selectItem ( COMMA selectItem )* -> ( selectItem )+ )
+			// SelectClauseParser.g:69:5: selectItem ( COMMA selectItem )*
+			{
+			pushFollow(FOLLOW_selectItem_in_selectList247);
+			selectItem7=selectItem();
+			state._fsp--;
+
+			stream_selectItem.add(selectItem7.getTree());
+			// SelectClauseParser.g:69:16: ( COMMA selectItem )*
+			loop5:
+			while (true) {
+				int alt5=2;
+				int LA5_0 = input.LA(1);
+				if ( (LA5_0==COMMA) ) {
+					alt5=1;
+				}
+
+				switch (alt5) {
+				case 1 :
+					// SelectClauseParser.g:69:18: COMMA selectItem
+					{
+					COMMA8=(Token)match(input,COMMA,FOLLOW_COMMA_in_selectList251);  
+					stream_COMMA.add(COMMA8);
+
+					pushFollow(FOLLOW_selectItem_in_selectList254);
+					selectItem9=selectItem();
+					state._fsp--;
+
+					stream_selectItem.add(selectItem9.getTree());
+					}
+					break;
+
+				default :
+					break loop5;
+				}
+			}
+
+			// AST REWRITE
+			// elements: selectItem
+			// token labels: 
+			// rule labels: retval
+			// token list labels: 
+			// rule list labels: 
+			// wildcard labels: 
+			retval.tree = root_0;
+			RewriteRuleSubtreeStream stream_retval=new RewriteRuleSubtreeStream(adaptor,"rule retval",retval!=null?retval.getTree():null);
+
+			root_0 = (CommonTree)adaptor.nil();
+			// 69:39: -> ( selectItem )+
+			{
+				if ( !(stream_selectItem.hasNext()) ) {
+					throw new RewriteEarlyExitException();
+				}
+				while ( stream_selectItem.hasNext() ) {
+					adaptor.addChild(root_0, stream_selectItem.nextTree());
+				}
+				stream_selectItem.reset();
+
+			}
+
+
+			retval.tree = root_0;
+
+			}
+
+			retval.stop = input.LT(-1);
+
+			retval.tree = (CommonTree)adaptor.rulePostProcessing(root_0);
+			adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+
+			 gParent.msgs.pop(); 
+		}
+
+		catch (RecognitionException e) {
+		  throw e;
+		}
+
+		finally {
+			// do for sure before leaving
+		}
+		return retval;
+	}
+	// $ANTLR end "selectList"
+
+
+	public static class selectTrfmClause_return extends ParserRuleReturnScope {
+		CommonTree tree;
+		@Override
+		public CommonTree getTree() { return tree; }
+	};
+
+
+	// $ANTLR start "selectTrfmClause"
+	// SelectClauseParser.g:72:1: selectTrfmClause : LPAREN selectExpressionList RPAREN inSerde= rowFormat inRec= recordWriter KW_USING StringLiteral ( KW_AS ( ( LPAREN ( aliasList | columnNameTypeList ) RPAREN ) | ( aliasList | columnNameTypeList ) ) )? outSerde= rowFormat outRec= recordReader -> ^( TOK_TRANSFORM selectExpressionList $inSerde $inRec StringLiteral $outSerde $outRec ( aliasList )? ( columnNameTypeList )? ) ;
+	public final HiveParser_SelectClauseParser.selectTrfmClause_return selectTrfmClause() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 196. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.